### PR TITLE
Improvements and fixes

### PR DESCRIPTION
**Pull request by kubecz3k**
Fixes two crashes and introduces ability to register post_process function

### CODE REVIEW

File: scene_synchronizer.h [hunk 1]
```diff
@@ -389,7 +392,7 @@ class ServerSynchronizer : public Synchronizer {
 	void generate_snapshot_node_data(const NetUtility::NodeData *p_node_data, bool p_force_full_snapshot, Vector<Variant> &r_result) const;
 };
 
-class ClientSynchronizer : public Synchronizer {
+class ClientSynchronizer : public Synchronizer, public Stringable {
```
Comment (kubecz3k) [inline scene_synchronizer.h:395]: oh no I need to get rid of that Stringable interface

File: scene_synchronizer.h [hunk 2]
```diff
@@ -441,6 +444,13 @@ class ClientSynchronizer : public Synchronizer {
 
 	void set_enabled(bool p_enabled);
 
+	String stringify(int tab_index) const override{
```
Comment (kubecz3k) [inline scene_synchronizer.h:447]: get rid of that function

File: scene_synchronizer.h [hunk 3]
```diff
@@ -39,6 +39,9 @@
 #include "net_utilities.h"
 #include <deque>
 
+#include "modules/rigid_netsync_investigation/stringable.h"
+#include "modules/rigid_netsync_investigation/rigid_netsync_investigation.h"
```
Comment (kubecz3k) [inline scene_synchronizer.h:43]: get rid of this

File: scene_synchronizer.cpp [hunk 4]
```diff
@@ -2437,7 +2464,7 @@ void ClientSynchronizer::process_controllers_recovery(real_t p_delta) {
 
 				const Variant current_val = nodes_to_recover[i]->vars[v].var.value;
 				nodes_to_recover[i]->vars[v].var.value = s_vars_ptr[v].value.duplicate(true);
-				node->set(s_vars_ptr[v].name, s_vars_ptr[v].value);
+				node->set(s_vars_ptr[v].name, s_vars_ptr[v].value); //rewinding, jgrzesik
```
Comment (kubecz3k) [inline scene_synchronizer.cpp:2467]: get rid of this comment

File: scene_synchronizer.cpp [hunk 5]
```diff
@@ -2306,7 +2332,7 @@ void ClientSynchronizer::process_controllers_recovery(real_t p_delta) {
 #ifdef DEBUG_ENABLED
 	// These are unreachable at this point.
 	CRASH_COND(server_snapshots.empty());
-	CRASH_COND(server_snapshots.front().input_id != checkable_input_id);
+	CRASH_COND(server_snapshots.front().input_id != checkable_input_id); //TODO log this, jgrzesik
```
Comment (kubecz3k) [inline scene_synchronizer.cpp:2335]: get rid of this comment

File: scene_synchronizer.cpp [hunk 6]
```diff
@@ -117,7 +117,7 @@ void SceneSynchronizer::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_rpc_notify_peer_status", "enabled"), &SceneSynchronizer::_rpc_notify_peer_status);
 
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "server_notify_state_interval", PROPERTY_HINT_RANGE, "0.001,10.0,0.0001"), "set_server_notify_state_interval", "get_server_notify_state_interval");
-	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "comparison_float_tolerance", PROPERTY_HINT_RANGE, "0.000001,0.01,0.000001"), "set_comparison_float_tolerance", "get_comparison_float_tolerance");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "comparison_float_tolerance", PROPERTY_HINT_RANGE, "0.0,0.01,0.000001"), "set_comparison_float_tolerance", "get_comparison_float_tolerance");
```
Comment (kubecz3k) [inline scene_synchronizer.cpp:120]: @AndreaCatania do we want to allow to have 0 tolerance here? It was helpful when debugging, not sure if should be a feature in the end

Comment (AndreaCatania) [reply scene_synchronizer.cpp:120]: Yes, leave it

**Comment by AndreaCatania** [comment]
Super seeded by: https://github.com/GodotNetworking/network_synchronizer/commit/d312fcfdc9489c849256ff1693f0f2f0eb2bb81e

**Pull request closed without merging**